### ISSUE TITLE
Label's text is not being rendered when exported to png file alone or after adding it as child widget to a parent layout

### ISSUE THREAD

**Issue by bartoszxkozlowski** [kivy/kivy#7881]
**Software Versions**
* Python: 3.10.0 (laptop) / 3.10.4 (Raspberry Pi 4)
* OS: Windows 10 / Debian GNU/Linux 11 (bullseye)
* Kivy: 2.1.0 / 2.1.0
* Kivy installation method: pip / pip

**Describe the bug**
Label widget is not rendered properly (or, in fact, at all) when it is being exported to a png file whether by itself or after it is added to some parent layout. Its texture boundaries are shown correctly when using the _showborder_ module but the text itself is not visible.

**Expected behavior**
Label's text is rendered and visible on the exported png file.

**To Reproduce**
```python
import kivy
from kivy.uix.label import Label

Label(text="test").export_to_png("test.png")
```
The output of that snippet is an empty 100x100px png file.

**Code and Logs and screenshots**
Default size (100x100px) Label widget with a "Hello World!" text set put on a FloatLayout with a showborder module with custom colors enabled:
![custom_showborder](https://user-images.githubusercontent.com/12623251/168252019-0846ff7e-d7cb-453e-b710-c4e01bc2d086.png)

**Additional context**
I am in the process of writing code for a function for creating an image based on a FloatLayout with some child Image and Label class widgets added. The child widgets are created dynamically with their sources (Image class widgets) and other properties read from a YAML file. I have been successful to add the Image class widgets to the layout but it went uphill when I tried to add a Label widget.

I tried every possible way to add a base Label widget to the layout, create a custom Label class and modify its properties and then add it, etc., etc. I have only been successful to achieve the Label widget to be rendered properly, by putting the script below at the beginning of the script file and then adding the loaded widget to layout:
```python
KV = '''
Label:
    text: "Hello World"
'''
label_from_string: Label = Builder.load_string(KV)
```

In any other way I tried I have failed to achieve the Label's text to be rendered after export. I'll be happy to provide more information if needed. Cheers!

**Comment by matham** [comment]
You're exporting it before the text is rendered.

**Comment by bartoszxkozlowski** [comment]
In my main script I have also tried to bind a callback to Label's canvas.ask_update method (according to [that guide from documentation](https://kivy.org/doc/stable/api-kivy.graphics.instructions.html)). The callback was called, canvas updated but the text was still not rendered after exporting as png (although the rectangle was rendered). 

I'd appreciate any explanation of how should I do that then or why shouldn't it be allowed to work that way (instead of just closing the issue 🙁). And yes, I also tried to schedule the export using the Clock's schedule_once method, but that resulted in my FloatLayout going haywire and its child widgets' dimensions and positions changing to some weird values. Although yes, the text was indeed rendered and visible on the png but it was not positioned where it was originally set. Any help is appreciated.
